modules/services/gpg-agent.nix

```diff
@@ -256,79 +252,112 @@ in {
+            };
+
+            Install = { WantedBy = [ "sockets.target" ]; };
+          };
+        })
+
+        (mkIf cfg.enableExtraSocket {
+          systemd.user.sockets.gpg-agent-extra = {
+            Unit = {
+              Description =
+                "GnuPG cryptographic agent and passphrase cache (restricted)";
+              Documentation = "man:gpg-agent(1) man:ssh(1)";
+            };
+
+            Socket = {
+              ListenStream = gpgconf "S.gpg-agent.extra";
+              FileDescriptorName = "extra";
+              Service = "gpg-agent.service";
+              SocketMode = "0600";
+              DirectoryMode = "0700";
+            };
+
+            Install = { WantedBy = [ "sockets.target" ]; };
+          };
+        })
+      ]))
+
+    (mkIf pkgs.stdenv.hostPlatform.isDarwin {
+      launchd.agents.gpg-agent = {
+        enable = true;
+        config = {
+          ProgramArguments = [ "${gpgPkg}/bin/gpg-connect-agent" "/bye" ];
+          RunAtLoad = cfg.enableSshSupport;
+          EnvironmentVariables = { GNUPGHOME = homedir; };
+          KeepAlive.SuccessfulExit = false;
         };
       };
-
-      systemd.user.sockets.gpg-agent = {
-        Unit = {
-          Description = "GnuPG cryptographic agent and passphrase cache";
-          Documentation = "man:gpg-agent(1)";
-        };
-
-        Socket = {
-          ListenStream = gpgconf "S.gpg-agent";
-          FileDescriptorName = "std";
-          SocketMode = "0600";
-          DirectoryMode = "0700";
-        };
-
-        Install = { WantedBy = [ "sockets.target" ]; };
-      };
-    }
-
-    (mkIf cfg.enableSshSupport {
-      systemd.user.sockets.gpg-agent-ssh = {
-        Unit = {
-          Description = "GnuPG cryptographic agent (ssh-agent emulation)";
-          Documentation =
-            "man:gpg-agent(1) man:ssh-add(1) man:ssh-agent(1) man:ssh(1)";
-        };
-
-        Socket = {
-          ListenStream = gpgconf "S.gpg-agent.ssh";
-          FileDescriptorName = "ssh";
-          Service = "gpg-agent.service";
-          SocketMode = "0600";
-          DirectoryMode = "0700";
-        };
-
-        Install = { WantedBy = [ "sockets.target" ]; };
-      };
     })
 
-    (mkIf cfg.enableExtraSocket {
-      systemd.user.sockets.gpg-agent-extra = {
-        Unit = {
-          Description =
-            "GnuPG cryptographic agent and passphrase cache (restricted)";
-          Documentation = "man:gpg-agent(1) man:ssh(1)";
+    (mkIf (pkgs.stdenv.hostPlatform.isDarwin && cfg.enableSshSupport) {
+      # see https://github.com/drduh/YubiKey-Guide#macos-1
+      launchd.agents.gpg-agent-symlink = {
+        enable = true;
+        config = {
+          ProgramArguments = [
+            "/bin/sh"
+            "-c"
+            "/bin/ln -sf $GNUPGHOME/S.gpg-agent.ssh $SSH_AUTH_SOCK"
+          ];
+          RunAtLoad = true;
+          EnvironmentVariables = { GNUPGHOME = homedir; };
+          # KeepAlive.SuccessfulExit = false;
```

Comment:
I uncommented it since it's pretty standard in background `launchd` configurations I've seen. I copied it over from the nix-darwin module initially and meant to double check its purpose before enabling.